LTHPasscodeViewController/LTHPasscodeViewController.h

```diff
@@ -314,6 +314,13 @@
  */
 - (void)showLockScreenWithAnimation:(BOOL)animated withLogout:(BOOL)hasLogout andLogoutTitle:(NSString*)logoutTitle;
 /**
+ @brief				Used for displaying the lock. The passcode view is added directly on the keyWindow.
+ @param superview   The superview where is to be presented, used to measure the center of the view.
+ @param hasLogout   Set to @c YES for a navBar with a Logout button, set to @c NO for no navBar.
+ @param logoutTitle The title of the Logout button.
```

Comment:
That looks pretty nice. The only problem is that the lock's frame is actually equal to the window it was added in, not the superview:

```objc
- (void)rotateAccordingToStatusBarOrientationAndSupportedOrientations {
// ...

[self setIfNotEqualTransform: transform
                           frame: self.view.window.bounds];
}

- (void)setIfNotEqualTransform:(CGAffineTransform)transform frame:(CGRect)frame {
// ...

self.view.frame = frame;
}
```

Here it's not really a problem because the size of the popup is big enough to not notice it's actually just a section, and also the popup has `clipsToBounds = true`. I don't really see a case where things would be different, but imagine `clipsToBounds` would be `false`: the lock would be covering the whole window.

I don't have a scenario like yours readily available, could you please check the frame of the lock, after it was presented, and compare it with the popup's view frame?

Theoretically, this should be changed:

```objc
[self setIfNotEqualTransform: transform
                           frame: self.view.window.bounds];
```

to

```objc
[self setIfNotEqualTransform: transform];
```

and use its superview's frame instead, here:

```objc
- (void)setIfNotEqualTransform:(CGAffineTransform)transform {
    CGRect frame = self.view.superview.frame;
    if(!CGAffineTransformEqualToTransform(self.view.transform, transform)) {
        self.view.transform = transform;
    }
    if(!CGRectEqualToRect(self.view.frame, frame)) {
        self.view.frame = frame;
    }
}
```